pkg/products/threescale/reconciler_test.go

```diff
@@ -192,6 +274,9 @@ func TestThreeScale(t *testing.T) {
 			}
 			status, err := tsReconciler.Reconcile(ctx, scenario.Installation, scenario.Product, scenario.FakeSigsClient, &quota.ProductConfigMock{}, scenario.Uninstall)
 			if err != nil {
+				if scenario.WantErr {
+					return
+				}
```

Comment:
This should be rewritten to be a more standard format. example
```go
if (err != nil) != scenario.wantErr {
    t.Fatalf("Error reconciling %s: %v", constants.ThreeScaleSubscriptionName, err)
}
```

Reply:
I wanted to make the test fail if unexpected error is returned. If an error is expected, it will move onto the next test case.
With the approach you suggested the cases where I'm expecting an error will not be caught by the if statement and additional assertions will be verified on the return value (`status` in this case) that shouldn't have been performed at all because there was an error.

Reply:
I will accept that explanation only because there would be the `scenario.Assert()` function call. It would be good to be checking the status of the function when it returned with an expected error.